lib/ruby-lint/constant_loader.rb

```diff
@@ -70,13 +77,37 @@ def run(ast)
     #
     def after_initialize
       @loaded = Set.new
+      @module_nesting = []
+      @registry ||= RubyLint.registry
```

Comment:
Because most callers does not pass a `:registry` argument to `ConstantLoader.new`, in which case this ensures that the attribute is set to the global registry. But in the test we pass a private registry.

It works within the scheme Iterator#initialize(options) + Iterator#after_initialize but maybe I am missing a better way to do it.